shared/studio/tabs/auth/state/index.tsx

```diff
@@ -107,6 +108,25 @@ export class AuthAdminState extends Model({
       return /^\d+$/.test(dur) ? null : "Invalid duration";
     }
   ),
+  draftAllowedRedirectUrls: createDraftAuthConfig(
+    "allowed_redirect_urls",
+    "std::str",
+    (urls) => {
+      if (urls === null) return null;
+
+      const urlList = urls.split("\n").filter((str) => str.trim() !== "");
+      if (urlList.length > 128) {
+        return "Too many URLs, maximum supported number of URLs is 128."
+      }
+
+      return null;
+    },
+    (urls) => {
+      if (urls === null) return "{}";
+      const urlList = urls.split("\n").filter((str) => str.trim() !== "");
+      return `{${urlList.map((u) => `'${u}'`).join(", ")}}`;
```

Comment:
```suggestion
      return `{${urlList.map((u) => JSON.stringify(u)).join(", ")}}`;
```
I think the `'` character is valid in urls, and we don't check the urls are valid anyway, so the urls need escaping.

Reply:
Oh, good call about checking the validity of the URLs, seems like something we should do in the validate function. Will fix that too.